src/map.cpp

```diff
@@ -354,6 +356,15 @@ Map::Map(int id, World *world)
 		this->world->timer.Register(event);
 	}
 
+	for (NPC* npc : this->npcs)
+	{
+		if (npc->ENF().type == ENF::Priest)
+		{
+			this->wedding = new Wedding(this, npc->index);
```

Comment:
In looking into how things behave with `$remap`, I think there is a bug where arena gets unloaded and never reloaded. `LoadArena` is only ever called from the constructor but `Map::Reload` calls both `Unload` and `Load`.

I think we probably want to add a `LoadArena` call to the `Reload` call, move this to a `LoadWedding` method, and also call that in the `Reload` method. If the map mutates and the NPC is removed then there will be a wedding object for a nonexistent NPC.

Reply:
Oh and also delete/clean up the wedding if the map mutates in the `Unload` call.

Reply:
Oops, forgot to reply to this one.

Anyway, good observations here.
I've tidied up the map lifecycle issues around weddings and arena.